src/main/java/org/apache/maven/plugins/install/InstallMojo.java

```diff
@@ -88,56 +77,88 @@
     @Component
     private ProjectInstaller installer;
 
+    private enum State
+    {
+        SKIPPED, INSTALLED, TO_BE_INSTALLED
+    }
+
+    private static final String INSTALL_PROCESSED_MARKER = InstallMojo.class.getName() + ".processed";
+
+    private void putState( State state )
+    {
+        getPluginContext().put( INSTALL_PROCESSED_MARKER, state.name() );
```

Comment:
here `getPluginContext()` and in other lines, eg in `getState` method - `session.getPluginContext(...)` - why different method is used for state?

Reply:
good question
because Mojo API has a direct `getPluginContext()` method to get the plugin context for the current project
but getting the plugin context for another project in the reactor is a different API that is provided in MavenSession: `MavenSession.getPluginContext( PluginDescriptor, MavenProject )`